File: Problems/0501.-0600./0561._Array_Partition/ArrayPartition1.py
#!/bin/python3
class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        n = len(nums)
    
        # To maximize the sum of pairs sort the array
        nums = sorted(nums, reverse = 0)
        
        n -= 1; 
        max_sum = 0
        while (n >= 0):
            max_sum += min(nums[n], nums[n-1])
            n -= 2
        return max_sum

        

File: Problems/0501.-0600./0561._Array_Partition/ArrayPartition1.java
//#!/bin/java
class Solution {
    public int arrayPairSum(int[] nums) {
        // Sort the array
        Arrays.sort(nums);
        
        // Hold the maximum sum
        int max_sum = 0;
        for (int i = nums.length-1; i > -1; i -= 2) {
            max_sum += Math.min(nums[i-1], nums[i]);
        }
        return max_sum;
    }
}
